MultiplayerMod/Game/Mechanics/DoorEvents.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using HarmonyLib;
+using MultiplayerMod.Core.Patch;
+using MultiplayerMod.Multiplayer.Objects;
+
+namespace MultiplayerMod.Game.Mechanics;
+
+[HarmonyPatch(typeof(Door))]
+public static class DoorToggleEvents {
+
+    public static event EventHandler<DoorStateChangedEventArgs> StateChanged;
```

Comment:
Can we use just Action here?
Since you are not using sender and itsn't typed anyway.
Also you will be able to avoid creating `Args` classes, just specify all of them in the event.

Reply:
Yep, I thought about that initially, but decided to stick with the official approach to see it in the wild. But seems like it's too exhaustive, yep let's use `Action<>` and make inner args if suitable.